lib/utf8utils.c

```diff
@@ -36,15 +36,6 @@ _is_character_unsafe(gunichar uchar, const gchar *unsafe_chars)
   return _strchr_optimized_for_single_char_haystack(unsafe_chars, (gchar) uchar) != NULL;
```

Comment:
Another optimization I see is that we could probably check `(uchar >= 0x7f)` (or `0x80`) in `_is_character_unsafe()`.